pandas/tests/io/test_parquet.py

```diff
@@ -212,28 +212,39 @@ def check_error_on_write(self, df, engine, exc):
             with tm.ensure_clean() as path:
                 to_parquet(df, path, engine, compression=None)
 
-    def check_round_trip(self, df, engine, expected=None,
-                         write_kwargs=None, read_kwargs=None,
-                         check_names=True):
+    def do_round_trip(self, df, path, engine_impl, expected=None,
+                      write_kwargs=None, read_kwargs=None,
+                      check_names=True):
+
         if write_kwargs is None:
-            write_kwargs = {}
+            write_kwargs = {'compression': None}
+
         if read_kwargs is None:
             read_kwargs = {}
-        with tm.ensure_clean() as path:
-            df.to_parquet(path, engine, **write_kwargs)
-            result = read_parquet(path, engine, **read_kwargs)
 
-            if expected is None:
-                expected = df
-            tm.assert_frame_equal(result, expected, check_names=check_names)
+        df.to_parquet(path, engine_impl, **write_kwargs)
+        actual = read_parquet(path, engine_impl, **read_kwargs)
 
-            # repeat
-            to_parquet(df, path, engine, **write_kwargs)
-            result = pd.read_parquet(path, engine, **read_kwargs)
+        if expected is None:
+            expected = df
 
-            if expected is None:
-                expected = df
-            tm.assert_frame_equal(result, expected, check_names=check_names)
+        tm.assert_frame_equal(expected, actual, check_names=check_names)
+
+    def check_round_trip(self, df, engine, expected=None,
+                         write_kwargs=None, read_kwargs=None,
+                         check_names=True):
+
```

Comment:
huh? why are you adding this?

Reply:
This method was refactoed, please see here https://github.com/pandas-dev/pandas/pull/19135/files/026ecc7ebb988b9178d5da44f03136817c1f42ae#diff-539eb41a29c68cc27818db4b480c2fd6L215 and my rational here https://github.com/pandas-dev/pandas/pull/19135/files/026ecc7ebb988b9178d5da44f03136817c1f42ae#r161376540